va/va_test.go

```diff
@@ -1005,7 +1005,7 @@ func TestCheckCAAFallback(t *testing.T) {
 	prob = va.checkCAA(ctx, core.AcmeIdentifier{Value: "bad-local-resolver.com", Type: "dns"})
 	test.Assert(t, prob != nil, "returned ProblemDetails was nil")
 	test.AssertEquals(t, prob.Type, probs.ConnectionProblem)
-	test.AssertEquals(t, prob.Detail, "server failure at resolver")
+	test.AssertEquals(t, prob.Detail, "DNS problem: query timed out looking up CAA for bad-local-resolver.com")
```

Comment:
Response from chat:
> that was required because I removed `ProblemDetailsFromDNSError` which was doing a `if dnsErr, ok := err.(*DNSError); ok` test, but since the error we were returning from the mock wasn't a pointer it would always fail back to `detailServerFailure` which was not correct